src/components/Calculator.js

```diff
@@ -1,35 +1,92 @@
+
+const Display = ({ inputs, whoChanged }) => {
+  let finaloutput = '';
+  if (!inputs.total && !inputs.operation && !inputs.next) finaloutput = '0';
+  else if (whoChanged.dot) {
+    if (!inputs.operation) finaloutput = inputs.next;
+    else {
+      finaloutput = !inputs.operation ? inputs.next : `${inputs.total || ''} ${inputs.operation || ''} ${inputs.next || ''}`;
+    }
+  } else {
+    finaloutput = `${inputs.total || ''} ${inputs.operation || ''} ${inputs.next || ''}`;
+  }
+  return (
+    <div className="span4col griditem2 weirdgray">
+      {finaloutput}
+    </div>
+  );
+};
 
-const Display = ({ inputs }) => (<div className="span4col griditem2 weirdgray">{ inputs }</div>);
 Display.propTypes = {
-  inputs: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
+  inputs: PropTypes.shape({
+    total: PropTypes.string,
+    next: PropTypes.string,
+    operation: PropTypes.string,
+  }).isRequired,
+  whoChanged: PropTypes.shape({
+    total: PropTypes.bool.isRequired,
+    next: PropTypes.bool.isRequired,
+    operation: PropTypes.bool.isRequired,
+    dot: PropTypes.bool.isRequired,
+  }).isRequired,
 };
 
-const Calculator = () => (
-  <div className="grid4x6">
-    <Display inputs={[4, 5, 8]} />
-    <div className="griditem softgray">AC</div>
-    <div className="griditem softgray">+/-</div>
-    <div className="griditem softgray">%</div>
-    <div className="griditem weirdorange">÷</div>
-    <div className="griditem softgray">7</div>
-    <div className="griditem softgray">8</div>
-    <div className="griditem softgray">9</div>
-    <div className="griditem weirdorange">*</div>
-    <div className="griditem softgray">4</div>
-    <div className="griditem softgray">5</div>
-    <div className="griditem softgray">6</div>
-    <div className="griditem weirdorange">-</div>
-    <div className="griditem softgray">1</div>
-    <div className="griditem softgray">2</div>
-    <div className="griditem softgray">3</div>
-    <div className="griditem weirdorange">+</div>
-    <div className="span2col griditem softgray">0</div>
-    <div className="griditem softgray">.</div>
-    <div className="griditem weirdorange">=</div>
-  </div>
-);
+const Calculator = () => {
+  const [inputs, setInputs] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const [whoChanged, setWhoChanged] = useState({
+    total: false,
+    next: false,
+    operation: false,
+    dot: false,
+  });
+
+  const handleButtonClick = (inputs, value) => {
+    const newInputs = calculate(inputs, value);
+    setWhoChanged({
+      total: inputs.total !== newInputs.total,
+      next: inputs.next !== newInputs.next,
+      operation: inputs.operation !== newInputs.operation,
+      dot: value === '.',
+    });
+    setInputs(newInputs);
+  };
+  handleButtonClick.propTypes = {
+    value: PropTypes.string.isRequired,
+  };
+
+  return (
+    <div className="grid4x6">
+      <Display inputs={inputs} whoChanged={whoChanged} />
```

Comment:
- [ OPTIONAL] Please fix this warning generated due to incorrect prop types.
![warn](https://github.com/EmiLoKiJu/my-app/assets/99656687/1f6fff98-2e52-482b-9994-a5ac5c839215)